config/backend-config/backend-config.go

```diff
@@ -39,9 +39,13 @@ var (
 
 	//DefaultBackendConfig will be initialized be Setup to either a WorkspaceConfig or MultiWorkspaceConfig.
 	DefaultBackendConfig BackendConfig
+	Http                 sysUtils.HttpI           = sysUtils.NewHttp()
+	log                  logger.LoggerI           = logger.NewLogger()
+	Ioutil               sysUtils.IoUtilI         = sysUtils.NewIoUtil()
+	Diagnostics          diagnostics.DiagnosticsI = diagnostics.NewDiagnostics()
```

Comment:
If we are going to mock these, instead of setting them to a mock before and resetting them to original after, is it not a good idea to encapsulate these in a struct and use that for testing?

Reply:
I have gone through the code a little bit more and I found my answer. Marking this resolved